x/subscription/keeper/subscription.go

```diff
@@ -401,6 +407,121 @@ func (k Keeper) resetSubscriptionDetailsAndAppendEntry(ctx sdk.Context, sub *typ
 	}
 }
 
+func (k Keeper) applyPlanDiscountIfEligible(duration uint64, plan *planstypes.Plan, price *sdk.Coin) {
+	if duration >= MONTHS_IN_YEAR {
+		// adjust cost if discount given
+		discount := plan.GetAnnualDiscountPercentage()
+		if discount > 0 {
+			factor := int64(100 - discount)
+			price.Amount = price.Amount.MulRaw(factor).QuoRaw(100)
+		}
+	}
+}
+
+func (k Keeper) chargeFromCreatorAccountToModule(ctx sdk.Context, creator sdk.AccAddress, price sdk.Coin) error {
+	if k.bankKeeper.GetBalance(ctx, creator, k.stakingKeeper.BondDenom(ctx)).IsLT(price) {
+		return utils.LavaFormatWarning("create subscription failed", legacyerrors.ErrInsufficientFunds,
+			utils.LogAttr("creator", creator),
+			utils.LogAttr("price", price),
+		)
+	}
+
+	err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, creator, types.ModuleName, []sdk.Coin{price})
+	if err != nil {
+		return utils.LavaFormatError("create subscription failed. funds transfer failed", err,
+			utils.LogAttr("creator", creator),
+			utils.LogAttr("price", price),
+		)
+	}
+
+	return nil
+}
+
+func (k Keeper) CreateFutureSubscription(ctx sdk.Context,
+	creator string,
+	consumer string,
+	planIndex string,
+	duration uint64,
+) error {
+	var err error
+
+	block := uint64(ctx.BlockHeight())
+	creatorAcct, plan, err := k.verifySubscriptionBuyInput(ctx, block, creator, consumer, planIndex)
+	if err != nil {
+		return err
+	}
+
+	var sub types.Subscription
+	found := k.subsFS.FindEntry(ctx, consumer, block, &sub)
+
+	if !found {
+		return utils.LavaFormatWarning("could not find active subscription. advanced purchase is only available for an active subscription", nil,
+			utils.LogAttr("consumer", consumer),
+			utils.LogAttr("block", block),
+		)
+	}
+
+	createFutureSubscription := func(sub *types.Subscription, creator, planIndex string, planBlock, duration uint64) {
+		sub.FutureSubscription = &types.FutureSubscription{
+			Creator:        creator,
+			PlanIndex:      planIndex,
+			PlanBlock:      planBlock,
+			DurationBought: duration,
+		}
+	}
+
+	if sub.FutureSubscription != nil {
```

Comment:
var price
if future != nil {
    price = diff
} else
{
    price = price*months
}

simplifies the code

Reply:
Fixed in 45d5d61e806e6281af3e924463f1f2a09a40a984